### DISCUSSION TITLE
Adding a table to custom page

### DISCUSSION THREAD

**Discussion by vv-ac** [filamentphp/filament#2367] [Help]
Hi all,

I have a custom page that holds multiple forms.
Now I want to have that page to have a table in there.
Using the following excerpt:
```
namespace App\Filament\Resources\PostResource\Pages;

use App\Filament\Resources\PostResource;
use Filament\Resources\Pages\Page;

use Illuminate\Database\Eloquent\Builder;

use Filament\Tables;

use Tables\Concerns\InteractsWithTable;

class ManagePost extends Page implements Tables\Contracts\HasTable
{
    protected function getTableQuery(): Builder
    {
        return Postrequirement::query();
    }
}
```

I left out a lot in there re forms, mounting and so on but tried to keep it comprehensive.

In the respective blade I simply inserted:
```
{{ $this->table }}
```

Now refreshing the page leads to:
```
Class App\Filament\Resources\PostResource\Pages\ManagePost contains 24 abstract methods and must therefore be declared abstract or implement the remaining methods (Filament\Tables\Contracts\HasTable::callTableColumnAction, Filament\Tables\Contracts\HasTable::deselectAllTableRecords, Filament\Tables\Contracts\HasTable::getAllTableRecordKeys, ...)
```

I read the docs on table builder twice and tried to comprehend what I am doing wrong. Nowhere in the class I define an "abstract ...".
I tried to switch a lot of things but none worked.

Any remarks on this?
Thanks!

**Comment by danharrin** [comment]
`use Tables\Concerns\InteractsWithTable;` needs to be inside the class, it is a trait.

**Reply by vv-ac** [reply]
Yup, though and tried that before, leaving me with:
```
Typed property App\Filament\Resources\PostResource\Pages\ManagePost::$cachedTableFilters must not be accessed before initialization
```

I do not use any filters on the table and need a "default" here.

**Reply by danharrin** [reply]
Yeah, that is a bug that should be submitted in an Issue.

**Reply by vv-ac** [reply]
Oh, that's kind of cool :-).
I added an issue and hopefully did it in the correct manner:
https://github.com/laravel-filament/filament/issues/2372

Thanks.

**Reply by vv-ac** [reply]
Works with the newest version, thank you!
